docs/automatic-installation.md

```diff
@@ -0,0 +1,119 @@
+# Automatic Installation
+
+Harvester can be installed in a mass manner. This document provides an example to do the automatic installation with PXE boot.
+
+## Preparing TFTP and HTTP servers
+
+TFTP or HTTP servers are required to serve boot files. Please ensure these servers are setup correctly before continuing.
+
+We recommend using [iPXE](https://ipxe.org/) to perform the network boot. It has more features than PXELINUX and is likely available in modern NIC cards. If NIC cards don't come with iPXE firmware, iPXE firmware images can be loaded from the TFTP server first. TFTP server is also used to serve iPXE scripts.
+
+Let's assume the HTTP server's IP is `10.100.0.10`, and the HTTP server serves `/usr/share/nginx/html/` folder at path `http://10.100.0.10/`.
+
+## Preparing boot files
+
+- Download the [Harvester ISO](https://github.com/rancher/harvester/tags).
+
+- Serve the ISO file.
+  
+  Copy or move the ISO file to an appropriate location so it can be downloaded via HTTP server. e.g.,
+
+  ```
+  sudo mkdir -p /usr/share/nginx/html/harvester/
+  sudo cp /path/to/harvester-amd64.iso /usr/share/nginx/html/harvester/
+  ```
+
+- Extract kernel and initrd file from the ISO file.
+
+  ```
+  mkdir /tmp/harvester-iso
+  mkdir /tmp/harvester-squashfs
+  sudo mount -o loop /path/to/harvester-amd64.iso /tmp/harvester-iso
+  sudo mount /tmp/harvester-iso/k3os/system/kernel/current/kernel.squashfs /tmp/harvester-squashfs
+
+  sudo cp /tmp/harvester-squashfs/vmlinuz /usr/share/nginx/html/harvester/
+  sudo cp /tmp/harvester-iso/k3os/system/kernel/current/initrd /usr/share/nginx/html/harvester/
+
+  sudo umount /tmp/harvester-squashfs
+  sudo umount /tmp/harvester-iso
+  rm -r /tmp/harvester-iso /tmp/harvester-squashfs
+  ```
+
+  Make sure the `vmlinuz` and `initrd` files can be downloaded from the HTTP server.
```

Comment:
Seems like we should release initrd and vmlinuz files so that users don't need to extract by themselves

Reply:
@gitlawr How to publish those two files? Could you provide a guide?

Reply:
They are built in k3os and skipped in the previous harvester release:
https://github.com/rancher/harvester-installer/blob/7bc6e259acd8a0fc6b3ac1f4dfe6405d5b6ef29c/k3os/images/output/01-full/Dockerfile#L18